WebService.Test/v1/Controllers/StatusControllerTest.cs

```diff
@@ -0,0 +1,99 @@
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.IotHub;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.StorageAdapter;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using WebService.Test.helpers;
+using Xunit;
+using Xunit.Abstractions;
+using SimulationModel = Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.Simulation;
+using StatusController = Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controllers.StatusController;
+
+namespace WebService.Test.v1.Controllers
+{
+    public class StatusControllerTest
+    {
+        private const string SIMULATION_ID = "1";
+
+        private readonly Mock<IPreprovisionedIotHub> preprovisionedIotHub;
+        private readonly Mock<IStorageAdapterClient> storage;
+        private readonly Mock<ISimulations> simulations;
+        private readonly Mock<ILogger> logger;
+        private readonly Mock<IServicesConfig> servicesConfig;
+        private readonly Mock<IDeploymentConfig> deploymentConfig;
+        private readonly Mock<IIotHubConnectionStringManager> connectionStringManager;
+        private readonly Mock<ISimulationRunner> simulationRunner;
+        private readonly StatusController target;
+
+        public StatusControllerTest(ITestOutputHelper log)
+        {
+            this.preprovisionedIotHub = new Mock<IPreprovisionedIotHub>();
+            this.storage = new Mock<IStorageAdapterClient>();
+            this.simulations = new Mock<ISimulations>();
+            this.logger = new Mock<ILogger>();
+            this.servicesConfig = new Mock<IServicesConfig>();
+            this.deploymentConfig = new Mock<IDeploymentConfig>();
+            this.connectionStringManager = new Mock<IIotHubConnectionStringManager>();
+            this.simulationRunner = new Mock<ISimulationRunner>();
+
+            this.target = new StatusController(
+                this.preprovisionedIotHub.Object,
+                this.storage.Object,
+                this.simulations.Object,
+                this.logger.Object,
+                this.servicesConfig.Object,
+                this.deploymentConfig.Object,
+                this.connectionStringManager.Object,
+                this.simulationRunner.Object);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public async Task GetTest()
+        {
+            // Arrange
+            SetupSimulationForRunner();
+
+            this.simulationRunner
+                .Setup(x => x.GetActiveDevicesCount())
+                .Returns(5);
+
+            // Act
+            var result = await this.target.Get();
+
+            // Assert
+            Assert.Equal(3, result.Properties.Count);
```

Comment:
which properties are you validating ? Can you add more checks ? And tests for other cases if different properties are expected ?

Reply:
Currently is validating `ActiveDeviceCount`. Will add more checks when add `telemetry metrics`.

Reply:
testing for the count will mean that this test will fail everytime we add a new property. The test should fail only if the "device count" value is missing